apps/condo/domains/user/schema/ConfirmPhone.js

```diff
@@ -0,0 +1,385 @@
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+const { normalizePhone } = require('@condo/domains/common/utils/phone')
+const { SENDER_FIELD, DV_FIELD } = require('@condo/domains/common/schema/fields')
+const { has, isEmpty } = require('lodash')
+
+const { 
+    CONFIRM_PHONE_ACTION_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_VERIFICATION_FAILED, 
+    CONFIRM_PHONE_SMS_CODE_MAX_RETRIES_REACHED, 
+} = require('@condo/domains/user/constants/errors')
+
+const { COUNTRIES, RUSSIA_COUNTRY } = require('@condo/domains/common/constants/countries')
+const { SMS_VERIFY_CODE_MESSAGE_TYPE } = require('@condo/domains/notification/constants')
+
+const { 
+    SMS_CODE_LENGTH, 
+    SMS_CODE_TTL, 
+    CONFIRM_PHONE_ACTION_EXPIRY, 
+    CONFIRM_PHONE_SMS_MAX_RETRIES,
+} = require('@condo/domains/user/constants/common')
+
+const whiteList = conf.SMS_WHITE_LIST ? JSON.parse(conf.SMS_WHITE_LIST) : {}
+
+const generateSmsCode = (phone) => {
+    if (has(whiteList, phone)) { // Emulate firebase white list for development - no real send sms
+        return Number(whiteList[phone])
+    }
+    return faker.datatype.number({ 
+        min: Math.pow(10, SMS_CODE_LENGTH - 1), // example 6 symbols:  min = 10^(6-1) = 100000
+        max: Math.pow(10, SMS_CODE_LENGTH) - 1, // max = 10^6-1 = 999999
+    })
+}
+
+const ConfirmPhoneAction = new GQLListSchema('ConfirmPhoneAction', {
+    fields: {
+        dv: DV_FIELD,
+        sender: SENDER_FIELD,
+        phone: {
+            schemaDoc: 'Phone. In international E.164 format without spaces',
+            type: Text,
+            kmigratorOptions: { null: true, unique: false },
+            hooks: {
+                resolveInput: ({ resolvedData }) => {
+                    return normalizePhone(resolvedData['phone'])
+                },
+            },
+        },
+        token: {
+            schemaDoc: 'Unique token to complete registration',
+            type: Text,
+            isUnique: true,
+            isRequired: true,
+        },
+        smsCode: {
+            schemaDoc: 'Last sms code sent to user',
+            type: Integer,
+            length: SMS_CODE_LENGTH,
+        },
+        smsCodeRequestedAt: {
+            schemaDoc: 'Time when sms code was requested',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        smsCodeExpiresAt: {
+            schemaDoc: 'Time when sms code becomes not valid',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        retries: {
+            schemaDoc: 'Number of times sms code input from user failed',
+            type: Integer,
+            defaultValue: 0,
+        },                
+        isPhoneVerified: {
+            schemaDoc: 'Phone verification flag. User verify phone by access to secret sms message',
+            type: Checkbox,
+            defaultValue: false,
+        },
+        requestedAt: {
```

Comment:
what the difference between this field and `smsCodeRequestedAt` field?

Reply:
The first "requested at" for the whole confirmPhoneAction - it's a time when the user starts registration
The second one for SMS code - is updated every time user asks for resending SMS